specifyweb/frontend/js_src/lib/components/Notifications/hooks.tsx

```diff
@@ -29,17 +28,20 @@ export function useNotificationsFetch({
   const [notifications, setNotifications] = React.useState<
     RA<GenericNotification> | undefined
   >(undefined);
+
+  const lastFetchDateRef = React.useRef<Date | undefined>(undefined);
+
   React.useEffect(() => {
     let pullInterval = INITIAL_INTERVAL;
-    let lastFetchedTimestamp: Date | undefined;
     let timeout: NodeJS.Timeout | undefined = undefined;
 
     const doFetch = (since = new Date()): void => {
       const startFetchTimestamp = new Date();
 
-      const url = addSinceToNotification
-        ? getSinceUrl(`/notifications/messages/`, since)
-        : `/notifications/messages/`;
+      const url =
+        lastFetchDateRef.current === undefined
+          ? `/notifications/messages/`
+          : getSinceUrl(`/notifications/messages/`, since);
```

Comment:
sorry, I didn't look closer at getSinceUrl before.
it doesn't make sense for `getSinceUrl` function to receive the baseUrl argument - the base URL should always stay the same.
it should only receive the `since` argument

---

And change the type of `since` argument to `Date | undefined`, so that you can do something like this in here:
```tsx
const url = getSinceUrl(lastFetchDateRef.current === undefined ? undefined : since)
```

benefit - no duplication of URL